case-lib/pipeline.sh

```diff
@@ -7,46 +7,39 @@ func_pipeline_export()
         dlogi "Topology file name is not specified, unable to run command: ${BASH_SOURCE[-1]}" && exit 1
     fi
 
-    local tplg="$1" opt="" sofcard=${SOFCARD:-0} cmd=""
+    local tplg="$1" ignore="" sofcard=${SOFCARD:-0}
 
     # got tplg_file, verify file exist
-    tplg_path=`func_lib_get_tplg_path "$tplg"`
+    tplg_path=$(func_lib_get_tplg_path "$tplg")
     [[ "$?" != "0" ]] && dloge "No available topology for pipeline export" && exit 1
     dlogi "${BASH_SOURCE[-1]} will use topology $tplg_path to run the test case"
 
-    shift
     # create filter option string
-    if [ $# -ne 0 ]; then
-        opt="-f"
-        while [ $# -ne 0 ]
-        do
-            opt=$opt" $1"
-            shift
-        done
-    fi
+    local filter="$2"
 
     # create block option string
     if [ ${#TPLG_IGNORE_LST[@]} -ne 0 ]; then
-        opt=$opt" -b"
-        for key in ${!TPLG_IGNORE_LST[@]}
+        for key in "${!TPLG_IGNORE_LST[@]}"
         do
             dlogi "Pipeline list to ignore is specified, will ignore '$key=${TPLG_IGNORE_LST[$key]}' in test case"
-            opt=$opt" $key:'${TPLG_IGNORE_LST[$key]}'"
+            ignore=$ignore" $key:'${TPLG_IGNORE_LST[$key]}'"
         done
     fi
 
-    cmd=$(echo sof-tplgreader.py $tplg_path $opt -s $sofcard -e)
+    cmd="sof-tplgreader.py $tplg_path -f \"$filter\" -b \"$ignore\" -s $sofcard -e"
 
     OLD_IFS="$IFS" IFS=$'\n'
     dlogi "Run command to get pipeline parameters"
     dlogc "$cmd"
-    for line in $(eval $cmd);
+    for line in $(eval "$cmd");
     do
-        eval $line
+        eval "$line"
     done
+
+    opt="-f \"$filter\" -b \"$ignore\""
```

Comment:
move to line 28?

Reply:
the opt here is used for output. no other funciton.

Reply:
I mean it is the same code when you load it, so here is duplicate with line 29